### PR TITLE
Price Feed Resilience: honor final useMovingAverage state in updateAsset cache selection

### PR DESCRIPTION
**Pull request by 0xJem**
## Summary
- add a regression test covering `updateStrategy = false` while `useMovingAverage` is set in params
- fix `OlympusPrice.v2.updateAsset()` to use resolved final state (`finalUseMA`) when selecting cache behavior
- align implementation with the documented `IPricev2.UpdateAssetParams` contract

## Audit Reference
- Addresses **L-09** audit issue: `updateAsset()` cache-selection path read `params_.useMovingAverage` even when `updateStrategy=false`.

## Why this matters
When `updateStrategy=false`, strategy fields are documented as ignored. Before this fix, `params_.useMovingAverage` could still change whether a single supplied observation updated `Variant.LAST`, causing state differences from an ignored field.

## Validation
- `pnpm install`
- `pnpm build`
- `forge test --match-path src/test/modules/PRICE.v2/updateAsset.t.sol`
- `pnpm run lint:check`

## Notes
- Regression reproduces on audited `price-feed-improvements` branch and passes with this fix.

## Summary by CodeRabbit

* **Bug Fixes**
  * Added overflow protection to price oracle calculations to prevent unsafe type conversions

* **Tests**
  * Added test coverage for price update edge cases, including moving average configurations and cached pricing scenarios with overflow validation

### CODE REVIEW

File: src/policies/price/ChainlinkOracleCloneable.sol [hunk 1]
```diff
@@ -155,8 +156,7 @@ contract ChainlinkOracleCloneable is IChainlinkOracle, IOraclePriceCache, Clone
         uint256 price = FullMath.mulDiv(assetPriceUsd, 10 ** _priceDecimals(), quotePriceUsd);
 
         roundId = cachedPrice.roundId;
-        /// forge-lint: disable-next-line(unsafe-typecast)
-        answer = int256(price);
+        answer = SafeCast.toInt256(price);
```
[the remark this answers is not in the document]
Comment (0xJem) [reply src/policies/price/ChainlinkOracleCloneable.sol:159]: Fixed in #240
